app/routes.py

```diff
@@ -1,2 +1,274 @@
-from flask import Blueprint
+import requests
+from dotenv import load_dotenv
+import os
+import datetime
+from sqlalchemy import desc
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request, Blueprint, make_response, jsonify
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
 
+load_dotenv()
+
+
+@tasks_bp.route("/", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "GET":
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title)
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(desc(Task.title))
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks:
+            if not task.completed_at:
+                is_complete = False
+            else:
+                is_complete = True
+            tasks_response.append({
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": is_complete
+            })
+        return make_response(jsonify(tasks_response), 200)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        keys = ['title', 'description', 'completed_at']
+        for key in keys:
+            if key not in request_body:
+                task_response = {"details": "Invalid data"}
+                return make_response(task_response, 400)
```

Comment:
👍 

Nice extensible way to check for fields in the request. Note that this would still let us create a task with an empty title (either empty string or explicitly set to null in the JSON body). We don't have a test that checks this, but it would be good to keep in mind.

Additionally, we should be doing similar checks when PUTting a task as well (even though there aren't tests for it). So we could also think about moving checks like this into validation helpers so that they are easier to reuse elsewhere.

We could even think about adding a class method to `Task` to create a new `Task` using the dictionary from a request body

```python
    @classmethod
    def from_dict(values):
        # create a new task, and set the model values from the values passed in
        # be sure to validate that all required values are present, we could return `None` or raise an error if needed
        return new_task
```